src/build_obb/build_obb.cpp

```diff
@@ -0,0 +1,54 @@
+#include <DagMC.hpp>
+#include "moab/ProgOptions.hpp"
+
+int main(int argc, char* argv[]) {
+
+  std::string dag_file;
+  std::string out_file;
+  bool verbose = false;
+
+  ProgOptions po("build_obb: A tool to prebuild your DAGMC OBB Tree");
+
+  po.addOpt<void>("verbose,v", "Verbose output", &verbose);
+  po.addRequiredArg<std::string>("dag_file", "Path to DAGMC file to proccess", &dag_file);
+  po.addOpt<std::string>("output,o", "Specify the output filename (default "")", &out_file);
+
+  po.addOptionHelpHeading("Options for loading files");
+
+  po.parseCommandLine(argc, argv);
+
+  // make new DagMC
+  moab::DagMC* DAG = new moab::DagMC();
+
+  moab::ErrorCode rval;
+
+  // sets the output filename if none specified
+  if (out_file == "") {
+    int pos = dag_file.find(".h5m");
+    out_file = dag_file.substr(0, pos) + "_obb.h5m";
```

Comment:
Might be worth it to check here if `pos` is `-1`, indicating that `.h5m` has not been found.